tests/bsim/bluetooth/mesh/src/test_provision.c

```diff
@@ -885,6 +977,65 @@ static void test_provisioner_pb_remote_client_parallel(void)
 	PASS();
 }
 
+/** @brief Test Provisioning procedure on Remote Provisioning client:
+ * - verify procedure timeouts on unresponsive unprovisioned device.
+ */
+static void test_provisioner_pb_remote_client_provision_timeout(void)
+{
+	uint16_t pb_remote_server_addr;
+	uint8_t uuid[16];
+	uint32_t link_close_wait_start;
+	struct bt_mesh_rpr_scan_status scan_status;
+
+	k_sem_init(&scan_sem, 0, 1);
+	k_sem_init(&link_open_sem, 0, 1);
+
+	provisioner_pb_remote_client_setup();
+	bt_mesh_test_cfg_set(NULL, 300);
+
+	/* Provision the 2nd device over PB-Adv. */
+	ASSERT_OK(provision_adv(1, &pb_remote_server_addr));
+
+	/* Provision the 3rd device over PB-Remote. */
+	struct bt_mesh_rpr_node srv = {
+		.addr = pb_remote_server_addr,
+		.net_idx = 0,
+		.ttl = 3,
+	};
+
+	rpr_cli.scan_report = rpr_scan_report_parallel;
+
+	/* Offset timeline of test to give some time to 3rd device to setup and disable scanning */
+	k_sleep(K_SECONDS(10));
+
+	memcpy(uuid, dev_uuid, 16);
+	uuid[6] = '0' + 2;
+	uuid_to_provision_remote = uuid;
+
+	LOG_INF("Starting scanning for an unprov device...");
+	ASSERT_OK(bt_mesh_rpr_scan_start(&rpr_cli, &srv, uuid, 5, 1, &scan_status));
+	ASSERT_EQUAL(BT_MESH_RPR_SUCCESS, scan_status.status);
+	ASSERT_EQUAL(BT_MESH_RPR_SCAN_SINGLE, scan_status.scan);
+	ASSERT_EQUAL(1, scan_status.max_devs);
+	ASSERT_EQUAL(5, scan_status.timeout);
+
+	ASSERT_OK(k_sem_take(&scan_sem, K_SECONDS(20)));
+
+	ASSERT_OK(bt_mesh_provision_remote(&rpr_cli, &srv, uuid, 0, prov_addr));
+	link_close_wait_start = k_uptime_get_32();
+	ASSERT_EQUAL(k_sem_take(&prov_sem, K_SECONDS(20)), -EAGAIN);
+	ASSERT_EQUAL((link_close_timestamp - link_close_wait_start) / MSEC_PER_SEC, 10);
```

Comment:
Will it be safer to reset `link_close_timestamp` to some invalid value (e.g. -1) before starting remote provisioning? `prov_link_close` will be called for any provisioning and `link_close_timestamp` can be a timestamp after RPR server device was provisioned.